brigade.js

```diff
@@ -1,5 +1,96 @@
-const { events } = require("brigadier");
+const { events, Job } = require("brigadier");
 
-events.on("push", function(e, project) {
-  console.log("received push for commit " + e.commit)
-})
\ No newline at end of file
+events.on("pull_request", async function(e, project) {
+
+    // The payload is a string we need to parse into JSON to access
+    const ghData = JSON.parse(e.payload);
+
+    // Notify GH that 
+    var ghn = new Notification("dco", e, project);
+    ghn.text = "Checking for DCO";
+    await ghn.run();
+
+    // Get the commits to iterate over
+    var cj = new Job(`github-pr-commits`, "mattfarina/github-pr-commits:latest");
+    cj.env = {
+        GITHUB_REPO: project.repo.name,
+        GITHUB_TOKEN: project.secrets.ghToken,
+        GITHUB_PR_NUMBER: ghData.number.toString(),
+    }
+    res = await cj.run();
+    var tempJson = JSON.stringify(res.toString());
+    tempJson = tempJson.replace(/\\n/g, "\\n")
+                .replace(/\\'/g, "\\'")
+                .replace(/\\"/g, '\\"')
+                .replace(/\\&/g, "\\&")
+                .replace(/\\r/g, "\\r")
+                .replace(/\\t/g, "\\t")
+                .replace(/\\b/g, "\\b")
+                .replace(/\\f/g, "\\f");
+    const commits = JSON.parse(tempJson);
+    const re = /^Signed-off-by: (.*) <(.*)>$/im
+
+    var count = 0, missed = 0;
```

Comment:
what if instead of using `missed` as an int, you created an array, and did a `missed.push(commit.id)` or whatever that field is) so that in the `text` you could print a list of all the commit hashes that didn't have a "Signed-off-by" on them.

Reply:
That would work really well for the checks API. For general status notifications the descriptions is supposed to be "A short description of the status."

A list of shas could get quite long.

Oh the wonderful things I'm thinking of with the checks API.